Source/Game.cs

```diff
@@ -8,360 +8,371 @@ namespace Celeste64;
         imGuiRenderer.RebuildFontAtlas();
 
-		scenes.Push(new Startup());
-	}
+        scenes.Push(new Startup());
+    }
 
-	public override void Shutdown()
-	{
+    public override void Shutdown()
+    {
         TASMod.Deinitialize();
 
-		if (scenes.TryPeek(out var topScene))
-			topScene.Exited();
+        if (scenes.TryPeek(out var topScene))
+            topScene.Exited();
 
-		while (scenes.Count > 0)
-		{
-			var it = scenes.Pop();
-			it.Disposed();
-		}
+        while (scenes.Count > 0)
+        {
+            var it = scenes.Pop();
+            it.Disposed();
+        }
 
-		scenes.Clear();
-		instance = null;
-	}
+        scenes.Clear();
+        instance = null;
+    }
 
-	public bool IsMidTransition => transitionStep != TransitionStep.None;
+    public bool IsMidTransition => transitionStep != TransitionStep.None;
 
-	public void Goto(Transition next)
-	{
-		Debug.Assert(
-			transitionStep == TransitionStep.None ||
-			transitionStep == TransitionStep.FadeIn);
-		transition = next;
-		transitionStep = scenes.Count > 0 ? TransitionStep.FadeOut : TransitionStep.Perform;
-		transition.ToBlack?.Restart(transitionStep != TransitionStep.FadeOut);
+    public void Goto(Transition next)
+    {
+        Debug.Assert(
+            transitionStep == TransitionStep.None ||
+            transitionStep == TransitionStep.FadeIn);
+        transition = next;
+        transitionStep = scenes.Count > 0 ? TransitionStep.FadeOut : TransitionStep.Perform;
+        transition.ToBlack?.Restart(transitionStep != TransitionStep.FadeOut);
 
-		if (transition.StopMusic)
-			Music.Stop();
-	}
+        if (transition.StopMusic)
+            Music.Stop();
+    }
 
     private static MethodInfo m_Input_Step = typeof(Input).GetMethod("Step", BindingFlags.Static | BindingFlags.NonPublic) ?? throw new Exception("Input missing Step");
-	public override void Update()
-	{
+
+    public override void Update()
+    {
         TASMod.Update();
-        if (Manager.IsPaused()) return;
-
-		// update top scene
-		if (scenes.TryPeek(out var scene))
-		{
-			var pausing =
-				transitionStep == TransitionStep.FadeIn && transition.FromPause ||
-				transitionStep == TransitionStep.FadeOut && transition.ToPause;
-
-			if (!pausing)
-				scene.Update();
-		}
-
-		// handle transitions
-		if (transitionStep == TransitionStep.FadeOut)
-		{
-			if (transition.ToBlack == null || transition.ToBlack.IsFinished)
-			{
-				transitionStep = TransitionStep.Hold;
-			}
-			else
-			{
-				transition.ToBlack.Update();
-			}
-		}
-		else if (transitionStep == TransitionStep.Hold)
+        if (Manager.IsPaused())
+        {
+            if (!scenes.TryPeek(out var camscene) || camscene is not World world) return;
+            var player = world.Get<Player>();
+            if (player == null) return;
+            world.prevMousePosition = world.nextMousePosition;
+            world.nextMousePosition = Input.Mouse.Position;
+            player.Update();
+            player.LateUpdate();

```

Comment:
I don't really like the idea of still updating the player while paused..
But I understand that it's difficult to move the camera logic out of the player, so I'm fine with this for now.
However add a TODO comment about that.